src/GitHub.Api/Git/Repository.cs

```diff
@@ -107,7 +178,7 @@ public ITask Pull()
 
         public ITask Push()
         {
-            return repositoryManager.Push(CurrentRemote.Value.Name, CurrentBranch?.Name);
+            return repositoryManager.Push(CurrentRemote.Value.Name, CurrentBranch?.Name).Then(UpdateGitStatus);
```

Comment:
Should put the `Then` bit on a new line to make sure it's easy to read that `Push` has a significant side effect when it's done.

Reply:
👍